llvm/include/llvm/CodeGen/BasicTTIImpl.h

```diff
@@ -996,7 +996,17 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
         return TTI::SK_Splice;
       break;
     }
-    case TTI::SK_Select:
+    case TTI::SK_Select: {
+      int NumSubElts;
+      if (Mask.size() > 2 && ShuffleVectorInst::isInsertSubvectorMask(
+                                 Mask, NumSrcElts, NumSubElts, Index)) {
+        if (Index + NumSubElts > NumSrcElts)
+          return Kind;
+        SubTy = FixedVectorType::get(Ty->getElementType(), NumSubElts);
+        return TTI::SK_InsertSubvector;
+      }
+      break;
+    }
```

Comment:
I think better to fix the order in TargetTransformInfoImpl.h::getInstructionCost